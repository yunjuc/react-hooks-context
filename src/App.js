import React from 'react'
import TodoList from './components/TodoList'
import { TodoContextProvider } from './contexts/TodoContext'

function App() {
  return (
    <div className="App">
      <h1>ToDo App</h1>
      <TodoContextProvider>
        <TodoList />
      </TodoContextProvider>
    </div>
  )
}

export default App;
