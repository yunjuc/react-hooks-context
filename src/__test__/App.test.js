import React from 'react'
import App from '../App'
import { render, fireEvent, screen } from '@testing-library/react'

describe("App component", () => {
  it('App component is rendered', () => {
    const { container } = render(<App />)

    expect(container).toMatchSnapshot()
  });

  it('title is presented', () => {
    const { container } = render(<App />)
    const title = container.querySelector('h1')

    expect(title).toHaveTextContent('ToDo App')
  });
})
