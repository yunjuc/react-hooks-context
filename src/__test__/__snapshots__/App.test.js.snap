// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App component is rendered 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      ToDo App
    </h1>
    <div
      class="todo-container"
    >
      <form>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline"
          required=""
        >
          <input
            class="MuiInputBase-input MuiInput-input"
            placeholder="Add todo item"
            required=""
            type="text"
            value=""
          />
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          id="add"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Add
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
      <div
        class="MuiGrid-root"
      />
    </div>
  </div>
</div>
`;
